nmigen/hdl/rec.py

```diff
@@ -146,7 +144,18 @@ def concat(a, b):
                                                      src_loc_at=1 + src_loc_at)
 
     def __getattr__(self, name):
-        return self[name]
+        # must check `getattr` before `self` - we need to hit Value methods before fields
+        try:
+            # it seems we lose the implicit `self` argument at some point, so curry manually
+            res = getattr(Value, name)
```

Comment:
Maybe something like
```suggestion
            value_attr = getattr(Value, name)
```
?